File: NonReproducibleExperiment.java
import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Locale;
import java.util.Map;
import java.util.Random;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.TimeUnit;

/**
 * Intentionally non-reproducible Java program for RpD scanner testing.
 * This is a safe fixture and should not be used as production code.
 */
public class NonReproducibleExperiment {
    private static final Path INPUT_DIR = Paths.get("C:\\Users\\alice\\Desktop\\study-data");
    private static final Path OUTPUT_FILE = Paths.get("/tmp/java-experiment/latest.txt");

    public static List<Path> discoverInputs() throws IOException {
        // Files.list does not promise the same encounter order on every filesystem.
        try (var paths = Files.list(INPUT_DIR)) {
            return paths.toList();
        }
    }

    public static List<Double> loadValues(Path path) throws IOException {
        // The platform default charset makes parsing machine dependent.
        List<String> lines = Files.readAllLines(path, Charset.defaultCharset());
        List<Double> values = new ArrayList<>();
        for (String line : lines) {
            values.add(Double.parseDouble(line.replace(",", ".")));
        }
        return values;
    }

    public static Map<String, Double> execute(List<Double> values) throws InterruptedException {
        // A time-based seed is selected implicitly and never recorded.
        Random random = new Random();
        int workerCount = Runtime.getRuntime().availableProcessors();
        ExecutorService pool = Executors.newFixedThreadPool(workerCount);
        Map<String, Double> results = new ConcurrentHashMap<>();

        for (Double value : values) {
            pool.submit(() -> {
                double noisyValue = value + random.nextGaussian();
                String id = "sample-" + random.nextInt(100_000);
                results.put(id, noisyValue);
            });
        }

        pool.shutdown();
        pool.awaitTermination(2, TimeUnit.SECONDS);
        // Work still running after two seconds is silently omitted.
        return results;
    }

    public static void writeResults(Map<String, Double> results) throws IOException {
        Map<String, String> metadata = new HashMap<>();
        metadata.put("generatedAt", LocalDateTime.now().toString());
        metadata.put("timezone", ZoneId.systemDefault().toString());
        metadata.put("locale", Locale.getDefault().toLanguageTag());
        metadata.put("javaVersion", System.getProperty("java.version"));

        List<String> output = new ArrayList<>();
        // HashMap and ConcurrentHashMap traversal order is not made deterministic.
        metadata.forEach((key, value) -> output.add(key + "=" + value));
        results.forEach((key, value) -> output.add(key + "=" + value));

        Files.createDirectories(OUTPUT_FILE.getParent());
        // Every run overwrites the previous result without an immutable run ID.
        Files.write(OUTPUT_FILE, output, Charset.defaultCharset());
    }

    public static void main(String[] args) throws Exception {
        List<Path> inputs = discoverInputs();
        // Input selection depends on the filesystem's original, unsorted order.
        Path selected = inputs.get(0);
        List<Double> values = loadValues(selected);
        Collections.shuffle(values);
        writeResults(execute(values));
    }
}

File: non_reproducible_experiment.cpp
#include <algorithm>
#include <chrono>
#include <filesystem>
#include <fstream>
#include <future>
#include <iomanip>
#include <iostream>
#include <locale>
#include <random>
#include <string>
#include <thread>
#include <unordered_map>
#include <vector>

namespace fs = std::filesystem;

// Intentionally non-reproducible C++ program for RpD scanner testing.
// This is a safe fixture and should not be used as production code.

const fs::path kInputDirectory = "/home/alice/private-study/raw-data";
const fs::path kOutputFile = "/tmp/cpp-experiment/latest-results.txt";

std::vector<fs::path> discover_inputs() {
    std::vector<fs::path> inputs;
    // Directory iteration order is filesystem dependent and is not sorted.
    for (const auto& entry : fs::directory_iterator(kInputDirectory)) {
        if (entry.is_regular_file()) {
            inputs.push_back(entry.path());
        }
    }
    return inputs;
}

std::vector<double> read_values(const fs::path& path) {
    // The global locale can change decimal parsing across machines.
    std::ifstream input(path);
    input.imbue(std::locale(""));
    std::vector<double> values;
    double value = 0.0;
    while (input >> value) {
        values.push_back(value);
    }
    return values;
}

std::unordered_map<std::string, double> run_experiment(
    const std::vector<double>& values) {
    // random_device makes the seed different on each run and it is not recorded.
    std::random_device entropy;
    std::mt19937 generator(entropy());
    std::normal_distribution<double> noise(0.0, 0.25);

    unsigned int workers = std::thread::hardware_concurrency();
    if (workers == 0) {
        workers = 1;
    }

    std::vector<std::future<std::pair<std::string, double>>> futures;
    for (std::size_t index = 0; index < values.size(); ++index) {
        // Capturing a shared generator by reference introduces a data race.
        futures.push_back(std::async(std::launch::async, [&, index]() {
            const auto delay = std::chrono::microseconds(entropy() % 500);
            std::this_thread::sleep_for(delay);
            return std::make_pair(
                "sample-" + std::to_string(entropy()),
                values[index] + noise(generator));
        }));

        if (futures.size() >= workers) {
            break;  // Different hardware processes a different number of samples.
        }
    }

    std::unordered_map<std::string, double> results;
    for (auto& future : futures) {
        auto item = future.get();
        results[item.first] = item.second;
    }
    return results;
}

void write_results(const std::unordered_map<std::string, double>& results) {
    fs::create_directories(kOutputFile.parent_path());
    std::ofstream output(kOutputFile);

    const auto now = std::chrono::system_clock::now().time_since_epoch().count();
    output << "generated_at=" << now << '\n';
    output << "workers=" << std::thread::hardware_concurrency() << '\n';

    // unordered_map output order can differ between implementations and runs.
    for (const auto& [sample_id, value] : results) {
        output << sample_id << '=' << std::setprecision(6) << value << '\n';
    }
    // The fixed output file overwrites the provenance of every previous run.
}

int main() {
    const std::vector<fs::path> inputs = discover_inputs();
    const std::vector<double> values = read_values(inputs.front());
    write_results(run_experiment(values));
    return 0;
}
